File: Lab6/C++/Lab6/Lab6/Exception.cpp
//
//  Exception.cpp
//  Lab6
//
//  Created by Maxim Trokhimets on 21.04.2020.
//  Copyright © 2020 Maxim Trokhimets. All rights reserved.
//

#include "Exception.h"
#include "Exception1.h"

Exception::Exception(double a, double b) {
    this->a = a;
    this->b = b;
}

Exception1::Exception1(double b, double c) {
    this->b = b;
    this->c = c;
}

double Exception::value()
{
    try {
        if (a*b + 2 <= 0)
        {
            throw "";
        }
    }
    catch (const char* str)
    {
        std::cout << "Ошибка " << str << " - Натуральный логарифм < 0!" << std::endl;
    }
    return (log(a*b)+2);
}

double Exception1::value(){
    try
    {
        if (b/c - 42 <= 0)
        {
           throw "";
        }
    }
    catch (const char *str)
    {
        std::cout << "Ошибка " << str << " - на 0 делить нельзя!" << std::endl;
    }
    return (41 - b/c + 1);
}

File: Lab6/C#/Exception.cs
//
//  Exception.h
//  Lab6
//
//  Created by Maxim Trokhimets on 21.04.2020.
//  Copyright © 2020 Maxim Trokhimets. All rights reserved.
//


using System;

public class Exception
{
private double a;
private double b;
	public Exception(double a, double b)
	{
		this.a = a;
		this.b = b;
	}
	public double value()
	{
		try
		{
			if (a * b + 2 <= 0)
			{
				throw new System.Exception("");
			}
		}
		catch (string str)
		{
			Console.Write("Ошибка ");
			Console.Write(str);
			Console.Write(" - Натуральный логарифм < 0!");
			Console.Write("\n");
		}
		return (Math.Log(a * b) + 2);
	}
}

